usecases/utils_test.go

```diff
@@ -0,0 +1,164 @@
+package usecases
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/sdoque/mbaigo/components"
+)
+
+// mockTransport is used for replacing the default network Transport (used by
+// http.DefaultClient) and it will intercept network requests.
+type mockTransport struct {
+	respFunc func() *http.Response
+	hits     int
+	err      error
+}
+
+func newMockTransport(respFunc func() *http.Response, v int, err error) mockTransport {
+	t := mockTransport{
+		respFunc: respFunc,
+		hits:     v,
+		err:      err,
+	}
+	// Hijack the default http client so no actual http requests are sent over the network
+	http.DefaultClient.Transport = t
+	return t
+}
+
+// RoundTrip method is required to fulfil the RoundTripper interface (as required by the DefaultClient).
+// It prevents the request from being sent over the network, and count how many times
+// a http request was sent
+func (t mockTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	t.hits -= 1
+	//log.Printf("hits: %d", t.hits)
+	if t.hits == 0 {
+		return nil, t.err
+	}
+	resp = t.respFunc()
+	resp.Request = req
+	return resp, nil
+}
+
+// A mocked UnitAsset used for testing
+type mockUnitAsset struct {
+	Name        string              `json:"name"`    // Must be a unique name, ie. a sensor ID
+	Owner       *components.System  `json:"-"`       // The parent system this UA is part of
+	Details     map[string][]string `json:"details"` // Metadata or details about this UA
+	ServicesMap components.Services `json:"-"`
+	CervicesMap components.Cervices `json:"-"`
+}
+
+func (mua mockUnitAsset) GetName() string {
+	return mua.Name
+}
+
+func (mua mockUnitAsset) GetServices() components.Services {
+	return mua.ServicesMap
+}
+
+func (mua mockUnitAsset) GetCervices() components.Cervices {
+	return mua.CervicesMap
+}
+
+func (mua mockUnitAsset) GetDetails() map[string][]string {
+	return mua.Details
+}
+
+func (mua mockUnitAsset) Serving(w http.ResponseWriter, r *http.Request, servicePath string) {}
+
+// Create a error reader to break json.Unmarshal()
+type errReader int
+
+var errBodyRead error = fmt.Errorf("bad body read")
+
+func (errReader) Read(p []byte) (n int, err error) {
+	return 0, errBodyRead
+}
+func (errReader) Close() error {
+	return nil
+}
+
+// Variables used in testing
+var brokenUrl = string([]byte{0x7f})
+var errHTTP error = fmt.Errorf("bad http request")
+
+// Help function to create a test system
+func createTestSystem(ctx context.Context, broken bool) components.System {
+	// instantiate the System
```

Comment:
Declare `context.Background()` here, skip all other contexts and defer ctx.Close